src/compiler/crystal/syntax/parser.cr

```diff
@@ -3570,7 +3570,7 @@ module Crystal
         next_token_skip_space_or_newline
         block_arg = parse_block_arg(extra_assigns)
         skip_space_or_newline
-        if args.any?(&.name.==(block_arg.name)) || (found_double_splat && found_double_splat.name == block_arg.name)
+        if (!block_arg.name.empty? && args.any?(&.name.==(block_arg.name))) || (found_double_splat && found_double_splat.name == block_arg.name)
```

Comment:
Which is better condition, it or the following:

```crystal
        if !block_arg.name.empty? && (args.any?(&.name.==(block_arg.name)) || (found_double_splat && found_double_splat.name == block_arg.name))
```

It is same meaning because `found_double_splat` never be empty named.

Reply:
I prefer the current but maybe to extract both sides into two more locals?

```cr
duplicate_block_arg_name = !block_arg.name.empty? && args.any?(&.name.==(block_arg.name)
block_name_conflicts = found_double_splat && found_double_splat.name == block_arg.name
if duplicate_block_arg_name || block_name_conflicts